xcdat/dataset.py

```diff
@@ -155,6 +161,84 @@ def open_mfdataset(
     return ds
 
 
+def infer_or_keep_var(
+    dataset: xr.Dataset, data_var: Optional[str] = None
+) -> xr.Dataset:
+    """Infer the data variable(s) or keep a specific one in the Dataset.
+
+    If ``data_var`` is None, then this function checks the number of
+    regular (non-bounds) data variables in the Dataset. If there is a single
+    regular data var, then it will add an 'xcdat_infer' attr pointing to it in
+    the Dataset. XCDAT APIs can call `get_inferred_var()` to get the data var
```

Comment:
Add to docstring: "If there are multiple regular data vars, it returns the Dataset without setting the 'xcdat_infer' attr."